devito/ir/equations/algorithms.py

```diff
@@ -99,18 +102,30 @@ def lower_exprs(expressions, subs=None, **kwargs):
     --------
     f(x - 2*h_x, y) -> f[xi + 2, yi + 4]  (assuming halo_size=4)
     """
-    return _lower_exprs(expressions, subs or {})
+    return _lower_exprs(expressions, subs or {}, **kwargs)
 
 
-def _lower_exprs(expressions, subs):
+def _lower_exprs(expressions, subs, **kwargs):
+    # FIXME: Not sure why you can get this, but not access with index
+    sregistry = kwargs.get('sregistry')
+
     processed = []
+    rebuilt = {}
     for expr in as_tuple(expressions):
         try:
             dimension_map = expr.subdomain.dimension_map
         except AttributeError:
             # Some Relationals may be pure SymPy objects, thus lacking the subdomain
             dimension_map = {}
 
+        if sregistry and dimension_map:
```

Comment:
sregistry cannot be none except for perhaps some low-level tests, in which case you may just pass a dummy one instead